pkg/apis/provisioning/v0alpha1/types.go

```diff
@@ -51,9 +51,12 @@ type GitHubRepositoryConfig struct {
 	// By default, this is the main branch.
 	Branch string `json:"branch,omitempty"`
 
-	// Token for accessing the repository.
+	// Token for accessing the repository. If set, it will be encrypted into encryptedToken, then set to an empty string again.
 	// TODO: this should be part of secrets and a simple reference.
-	Token string `json:"token,omitempty"`
+	Token string `json:"token"`
```

Comment:
I think the omit empty is good in this case :) once encrypted no reason even to be in the responses.

Reply:
we must also check+clear kubectl annotations so the original plain text is not saved there